libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -510,10 +511,12 @@ export class BotFrameworkAdapter
         if (!reference.serviceUrl) {
             throw new Error('BotFrameworkAdapter.createConversation(): missing serviceUrl.');
         }
+        const logicParse = LogicT.safeParse(parametersOrLogic);
+        const parameterParse = conversationParametersObject.partial().safeParse(parametersOrLogic);
 
-        const parameters = LogicT.check(parametersOrLogic) ? {} : parametersOrLogic;
+        const parameters = parameterParse.success ? parameterParse.data : {};
 
-        const logic = LogicT.check(parametersOrLogic) ? parametersOrLogic : LogicT.parse(maybeLogic);
+        const logic = logicParse.success ? logicParse.data : LogicT.parse(maybeLogic);
```

Comment:
I think this is wrong, cant we just use .safeParse().success instead of .check().

Reply:
?? I'm not using check

Reply:
The thing is, that compared to previous implementation, the `parameters` variable should be

`const parameters = logicParse.success ? {} : logicParse.data;`

and for the logic is ok. Therefore the `parameterParse` and `conversationParametersObject` will no longer be necessary, right?

Reply:
I got it, no you can't that's why I assigned the parsed value (data) at the end, because check did the validation and converted the object to that type immediately or to the opposite type in case it failed. that's why I added those extra lines.

Reply:
Note that in the previous implementation, if the parse was successful with logicT, then an empty object was returned to parameters, if it failed, parametersOrLogic was returned. In fact, that's why that property is name that way, because it receives the logic OR the parameters.

Reply:
Ok, i think i got it.

The following code is added because in the `parametersOrLogic` variable could come two different types.
LogicT is a function type, so if the value is an object this will be false, and in the new conversationParametersObject, if it is a function it will be fails also.
`const parameterParse = conversationParametersObject.partial().safeParse(parametersOrLogic);`

Thats why the change.

Reply:
Yes, that is